pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesFunctionAuthProvider.java

```diff
@@ -30,7 +32,7 @@
  */
 public interface KubernetesFunctionAuthProvider extends FunctionAuthProvider {
 
-    void initialize(CoreV1Api coreClient, String kubeNamespace, byte[] caBytes);
+    void initialize(CoreV1Api coreClient, byte[] caBytes, java.util.function.Function<Function.FunctionDetails, String> namespaceCustomizerFunc);
```

Comment:
Now, instead of taking the namespace directly, it has a call to build the namespace from the `FunctionDetails`. This allows us to have it call the `KubernetesManifestCustomizer` via this function but doesn't require it to directly have knowledge of it.

Reply:
Since caBytes and namespaceCustomizerFunc are essentially optional arguments, lets add setters for them to be with default being a no-op so that the interface can be cleaner.  Not all implementations will require caBytes and namespaceCustomizerFunc.

Reply:
done